integration-cli/environment/clean.go

```diff
@@ -155,62 +152,47 @@ func deleteAllNetworks(t testingT, dockerBinary string, daemonPlatform string) {
 			// nat is a pre-defined network on Windows and cannot be removed
 			continue
 		}
-		status, b, err := request.SockRequest("DELETE", "/networks/"+n.Name, nil, request.DaemonHost())
+		err := c.NetworkRemove(context.Background(), n.ID)
 		if err != nil {
 			errs = append(errs, err.Error())
 			continue
 		}
-		if status != http.StatusNoContent {
-			errs = append(errs, fmt.Sprintf("error deleting network %s: %s", n.Name, string(b)))
-		}
 	}
 	if len(errs) > 0 {
 		t.Fatalf("%v", strings.Join(errs, "\n"))
 	}
 }
 
-func getAllNetworks() ([]types.NetworkResource, error) {
-	var networks []types.NetworkResource
-	_, b, err := request.SockRequest("GET", "/networks", nil, request.DaemonHost())
+func getAllNetworks(c *client.Client) ([]types.NetworkResource, error) {
+	networks, err := c.NetworkList(context.Background(), types.NetworkListOptions{})
 	if err != nil {
 		return nil, err
 	}
-	if err := json.Unmarshal(b, &networks); err != nil {
-		return nil, err
-	}
 	return networks, nil
 }
 
-func deleteAllPlugins(t testingT, dockerBinary string) {
-	plugins, err := getAllPlugins()
+func deleteAllPlugins(t testingT, c *client.Client, dockerBinary string) {
+	plugins, err := getAllPlugins(c)
 	if err != nil {
 		t.Fatalf("%v", err)
 	}
 	var errs []string
 	for _, p := range plugins {
-		pluginName := p.Name
-		status, b, err := request.SockRequest("DELETE", "/plugins/"+pluginName+"?force=1", nil, request.DaemonHost())
+		err := c.PluginRemove(context.Background(), p.Name, types.PluginRemoveOptions{Force: true})
 		if err != nil {
 			errs = append(errs, err.Error())
 			continue
 		}
-		if status != http.StatusOK {
-			errs = append(errs, fmt.Sprintf("error deleting plugin %s: %s", p.Name, string(b)))
-		}
 	}
 	if len(errs) > 0 {
 		t.Fatalf("%v", strings.Join(errs, "\n"))
 	}
 }
 
-func getAllPlugins() (types.PluginsListResponse, error) {
-	var plugins types.PluginsListResponse
-	_, b, err := request.SockRequest("GET", "/plugins", nil, request.DaemonHost())
+func getAllPlugins(c *client.Client) (types.PluginsListResponse, error) {
```

Comment:
I think this should be using `client.APIClient` (the interface), same is true in other places.